lib/fog/image/openstack/v2/models/images.rb

```diff
@@ -17,10 +17,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-              image_hash.merge(:service => service)
-            )
+            result = all.find { |image| image.id == id }
```

Comment:
@Carthaca whoops, misunderstood your original comment. Your fix worked perfectly, here's the [commit](https://github.com/fog/fog-openstack/pull/170/commits/bc4c198a4fdbd5180d03dde5bf286afea218556d)!